templates/_worker.js/routes-matcher.ts

```diff
@@ -297,6 +298,87 @@ export class RoutesMatcher {
 		return prevPath;
 	}
 
+	/**
+	 * Applies the route's redirects for locales and internationalization.
+	 *
+	 * @param route Build output config source route.
+	 */
+	private applyLocaleRedirects(route: VercelSource): void {
+		if (!route.locale?.redirect) return;
+
+		if (!this.locales) this.locales = {};
+		Object.assign(this.locales, route.locale.redirect);
+
+		// Automatic locale detection is only supposed to occur at the root. However, the build output
+		// sometimes uses `/` as the regex instead of `^/$`. So, we should check if the `route.src` is
+		// equal to the path if it is not a regular expression, to determine if we are at the root.
+		// https://nextjs.org/docs/pages/building-your-application/routing/internationalization#automatic-locale-detection
+		const srcIsRegex = /^\^(.)*$/.test(route.src);
+		if (!srcIsRegex && route.src !== this.path) return;
+
+		// If we already have a location header set, we might have found a locale redirect earlier.
+		if (this.headers.normal.has('location')) return;
+
+		const {
+			locale: { redirect: redirects, cookie: cookieName },
+		} = route;
+
+		const cookieValue = cookieName ? this.cookies[cookieName] ?? '' : '';
+		const cookieLocales = parseAcceptLanguage(cookieValue);
```

Comment:
nit:

```suggestion
		const cookieValue = cookieName && this.cookies[cookieName];
		const cookieLocales = parseAcceptLanguage(cookieValue ?? '');
```

(if you prefer your current version I am 100% fine with it, I was just thinking that the `?? '' : ''` can look a bit unclear :thinking:)

Reply:
Shouldn't that be `||` instead?

Reply:
the `&&` you mean?

I don't think so, if `cookieName` is falsy (say `undefined`) then we get:
`const cookieValue = undefined && this.cookies[undefined]` which results in `undefined`

if `cookieName` is truhy (say `'str'`) then we get:
`const cookieValue = 'str' && this.cookies['str']` which results in `this.cookies['str']`

isn't this correct? :thinking:

Reply:
The ?? not the &&

Reply:
lol sorry :sweat_smile: